adrs/8702033095/README.md

```diff
@@ -0,0 +1,56 @@
+# URL Path Casing and Separators
+
+* **State:** Reviewing
+* **Created:** 2021-12-13
+* **Tags:** seo, url, openapi
+
+## Context
+
+The lack of alignment in URL design across products causes multiple issues in
+software development.
+
+Suboptimal SEO where some Search Engines don't recognize underscores in URLs. A
+URL containing "my_page" will be indexed as "my page" (as one term) while a URL
+containing "my-page" will be indexed as "my" "page" (as two terms).
+
+Worth saying that this is primarily how Google Search Engine does things,
+regardless, it is safe to say that it is prudent following Google Search
+optimizations.
+
+Suboptimal user-experience since camelCase and underscore also require the user
+to use the shift key, whereas hyphenated does not on desktop keywords; and
+characters like underscore are hidden behind special symbols in mobile phones
+keywords.
+
+Lack of shared tooling, forcing developers to reimplement tools or promoting
+bikeshedding.
+
+## Resolution
+
+Creating "perfect software" is close to impossible. The following suggestions
+may feel like suggestions because we don't have the full scope of the situation.
+
+Therefore, you **MAY** follow the following resolutions unless you have a
+technical limitation. You **MUST** assumed such rules de facto in conversations.
+
+You **MUST** use the following characters anywhere in the path:
+
+* U+0061 to U+007A, "a-z"
+* U+0030 to U+0039, "0-9"
```

Comment:
What about `.` (period) which are used for things like file names "data.csv" or IDs sometimes and `~` which I like to use to separate out admin paths sometimes e.g. `mysite.com/~/settings`

Reply:
```
Characters that are allowed in a URI, but do not have a reserved purpose, are called unreserved. These include uppercase and lowercase letters, decimal digits, hyphen, period, underscore, and tilde.

  ALPHA  DIGIT  "-" / "." / "_" / "~"
```

Reply:
@tleef shit good catch ... forgot if `/hello.json` was part of the URL Path or not.